examples/ensemble/plot_gradient_boosting_early_stopping.py

```diff
@@ -0,0 +1,122 @@
+`n_estimators_`.
+
+This example illustrates how the early stopping can used in the
+:class:`sklearn.ensemble.GradientBoostingClassifier` model to achieve
+almost the same accuracy as compared to a model built without early stopping
+using much less estimators. This can save memory and prediction time.
+"""
+
+# Authors: Vighnesh Birodkar <vighneshbirodkar@nyu.edu>
+#          Raghav RV <rvraghav93@gmail.com>
+# License: BSD 3 clause
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn import ensemble
+from sklearn import datasets
+from sklearn.model_selection import train_test_split
+
+print(__doc__)
+
+data_list = [datasets.load_iris(), datasets.load_digits()]
+data_list = [(d.data, d.target) for d in data_list]
+data_list += [datasets.make_hastie_10_2()]
+names = ['Iris Data', 'Digits Data', 'Hastie Data']
+
+n_gb = []
+score_gb = []
+n_gbes = []
+score_gbes = []
+
+n_estimators = 500
+
+for X, y in data_list:
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2,
+                                                        random_state=0)
+
+    # We specify that if the scores don't improve by atleast 0.01 for the last
+    # 10 stages, stop fitting additional stages
+    gbes = ensemble.GradientBoostingClassifier(n_estimators=n_estimators,
+                                               validation_fraction=0.2,
+                                               n_iter_no_change=5, tol=0.01,
+                                               random_state=0)
+    gb = ensemble.GradientBoostingClassifier(n_estimators=n_estimators,
+                                             random_state=0)
+
+    gb.fit(X_train, y_train)
+    gbes.fit(X_train, y_train)
+
+    score_gb.append(gb.score(X_test, y_test))
+    score_gbes.append(gbes.score(X_test, y_test))
+
+    n_gb.append(gb.n_estimators_)
+    n_gbes.append(gbes.n_estimators_)
+
+plt.figure(figsize=(9, 5))
+bar_width = 0.2
+n = len(data_list)
+index = np.arange(0, n*bar_width, bar_width)*2.5
+index = index[0:n]
+
+plt.subplot(121)
+
+plt.bar(index, n_gb, bar_width, label='Without early stopping' , color='teal')
+plt.bar(index + bar_width, n_gbes, bar_width,
+        label='With early stopping', color='cyan')
+
+max_y = np.amax(np.maximum(n_gb, n_gbes))
+
+plt.xticks(index + bar_width, names)
+plt.yticks(np.arange(0, 530, 50))
+
+plt.ylim([0, 630])
+plt.legend(loc='best')
+plt.grid(True)
+
+plt.xlabel('Datasets')
+plt.ylabel('Number of Estimators')
+
+plt.subplot(122)
+
+plt.bar(index, score_gb, bar_width, label='Without early stopping',
+        color='crimson')
+plt.bar(index + bar_width, score_gbes, bar_width,
+        label='With early stopping', color='coral')
+max_y = np.amax(np.maximum(score_gb, score_gbes))
+
+plt.xticks(index + bar_width, names)
+plt.yticks(np.arange(0, 1.3, 0.1))
+
+plt.ylim([0, 1.3])
+plt.legend(loc='best')
+plt.grid(True)
+
+plt.xlabel('Datasets')
+plt.ylabel('Score')
```

Comment:
just show the scores and print n_estimators

try to keep the example short

can you paste the produced figure?